Week-02/word_counter.go

```diff
@@ -0,0 +1,113 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+	"sync"
+	"unicode"
+)
+
+const MAX_POOL = 2
+
+func getNextWord(file *os.File) (string, error) {
+	var buff = make([]byte, 1) // TODO: Optimize buffer size to improve reading performance
+	var stringBuilder []rune
+
+	for {
+		_, err := file.Read(buff)
+
+		if err == io.EOF && len(stringBuilder) > 0 {
+			return string(stringBuilder), nil
+		}
+
+		if err != nil {
+			return "", err
+		}
+
+		switch c := rune(buff[0]); {
+		case unicode.IsLetter(c):
+			fallthrough
+		case unicode.IsDigit(c):
+			stringBuilder = append(stringBuilder, c)
+		case c == ' ':
+			if len(stringBuilder) > 0 {
+				return string(stringBuilder), nil
+			}
+		}
+	}
+}
+
+func wordCounter(wg *sync.WaitGroup, files chan string, words chan string) {
+	// Do until files chanel close
```

Comment:
the best practice is, just to make sure you don't forget it at the end of this method
```
defer wg.Done()
```